src/chai/ManagedArray.inl

```diff
@@ -214,11 +247,26 @@ CHAI_HOST void ManagedArray<T>::reallocate(size_t elems)
 
 template<typename T>
 CHAI_INLINE
-CHAI_HOST void ManagedArray<T>::free()
+CHAI_HOST void ManagedArray<T>::free(ExecutionSpace space)
 {
-  if(!m_is_slice) {
-    m_resource_manager->free(m_pointer_record);
-    m_pointer_record = nullptr;
+  if(!m_is_slice && *this != nullptr) {
+    if (m_resource_manager == nullptr) {
+       m_resource_manager = ArrayManager::getInstance();
+    }
+    if (m_pointer_record == &ArrayManager::s_null_record) {
```

Comment:
this is just some recovery code - if somehow we get to this point, it may be because the application is out of sync with CHAI, and since the application is asking CHAI to free something, CHAI is going to go ahead and take ownership of it via the makeManaged and then free it. (Again, very rare if impossible to get to this state, but this is paranoia land)

Reply:
OK.